server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -701,7 +701,7 @@ public void initRestHandlers(Supplier<DiscoveryNodes> nodesInCluster) {
 
         registerHandler.accept(new RestIndexAction());
         registerHandler.accept(new CreateHandler());
-        registerHandler.accept(new AutoIdHandler(clusterService));
+        registerHandler.accept(new AutoIdHandler(nodesInCluster));
```

Comment:
compat_rest_api was updated to master where it was (accidentally I suspect) reverted back to use clusterService when nodes in cluster is only needed https://github.com/elastic/elasticsearch/commit/8264bdd36a2db3c204387c4f8c8a846001bee899
I suspect this will be back to nodesInCluster in master soon